src/closest_color.jl

```diff
@@ -3,18 +3,16 @@ Simplest form of image quantization by turning each pixel to the closest one
 in the provided color palette `cs`.
 Technically this not a dithering algorithm as the quatization error is not "randomized".
 """
-struct ClosestColor <: AbstractCustomColorDither end
+struct ClosestColor <: AbstractDither end
 
-function binarydither!(::ClosestColor, out::GenericGrayImage, img::GenericGrayImage)
-    return out .= img .> 0.5
+function binarydither(::ClosestColor, img::GenericGrayImage)
+    return map(px -> px > 0.5 ? INDEX_WHITE : INDEX_BLACK, img)
 end
 
-function colordither!(
-    ::ClosestColor,
-    out::GenericImage,
-    img::GenericImage,
-    cs::AbstractVector{<:Pixel},
-    metric::DifferenceMetric,
+function colordither(
+    ::ClosestColor, img::GenericImage, cs::AbstractVector{<:Pixel}, metric::DifferenceMetric
 )
-    return out .= eltype(out).(map((px) -> closest_color(px, cs; metric=metric), img))
+    cs = Lab{floattype(eltype(eltype(img)))}.(cs)
```

Comment:
Is there a more elegant way to do this? E.g. if I have a `Float{T}`, I want to have a `Lab{T}`.

Reply:
Maybe https://github.com/JuliaImages/juliaimages.github.io/issues/151 (I myself always forget about this)

```julia
julia> using ImageCore

julia> ccolor(Lab, RGB{N0f8})
Lab{Float32}
```